src/shell/element/surface.rs

```diff
@@ -124,6 +127,34 @@ impl CosmicSurface {
         }
     }
 
+    pub fn corner_radius(&self) -> Option<[u8; 4]> {
+        self.wl_surface().and_then(|surface| {
+            with_states(&surface, |s| {
+                let d = s
+                    .data_map
+                    .get::<Mutex<Weak<CosmicCornerRadiusToplevelV1>>>()?;
+                let guard = d.lock().unwrap();
+
+                let weak_data = guard.upgrade().ok()?;
+
+                let corners = weak_data.data::<CornerRadiusData>()?;
+
+                let guard = corners.lock().unwrap();
+
+                let size = self.0.bbox().size;
```

Comment:
This is not the correct size, the protocol mentions the xdg-toplevel geometry. So this should be `<CosmicSurface as SpaceElement>::geometry()` instead, I think.